### PR TITLE
Graph Switching Round 2

### PR DESCRIPTION
**Pull request by christinem**
This continues the graph switching feature. A second graph can now be properly parsed and added as a graph to switch to - with its nodes being interactive. As well, graph buttons are now dynamically generated by grabbing the JSON objects from the new 'graphs' page @Ian-Stewart-Binks added, to determine which graphs are currently parsed in the database. The hope is that this will allow it to be alot easier to add new graphs in the future. 

(Sorry for the extra commits that aren't mine, I had to merge in Ian's database changes).

### CODE REVIEW

File: public/js/graph/sidebar/sidebar_events.js [hunk 1]
```diff
@@ -51,25 +84,28 @@ $('#graph-csc').click(function(e) {
...
+        async: false,
+        success: function (data) {
+            graphs = data;
+        },
+        error: function () {
+            throw 'No graphs in database';
+        }
+    });
+
+    return graphs;
```
Comment (david-yz-liu) [inline public/js/graph/sidebar/sidebar_events.js]: This is the wrong order in which to do things. You should pass `graphs` as an argument to `createGraphButtons`, calling it inside the `success` function.

File: public/js/graph/tabs/focuses.js [hunk 2]
```diff
@@ -61,6 +61,9 @@ function spotlight(id) {
                                                            '" ry="',
                                                            String(height + 8.5),
                                                            '"/>');
+
+    console.log(ellipse);
```
Comment (david-yz-liu) [inline public/js/graph/tabs/focuses.js]: Looks like this is working, so you should remove these.

**Comment by david-yz-liu** [comment]
@christinem Good work; I had two comments, after which I'll merge this in.

**Comment by christinem** [comment]
@david-yz-liu Thanks, changes have been made!

File: public/js/graph/sidebar/sidebar_events.js [hunk 3]
```diff
@@ -51,25 +83,26 @@ $('#graph-csc').click(function(e) {
...
-
-});
\ No newline at end of file
+    var graphs;
+    $.ajax({
+        url: 'graphs',
+        dataType: 'json',
+        async: false,
+        success: function (data) {
+            graphs = data;
```
Comment (david-yz-liu) [inline public/js/graph/sidebar/sidebar_events.js]: There should be a `var` here

**Pull request merged**